classes/sota_raspberrypi.bbclass

```diff
@@ -16,25 +16,55 @@ DEV_MATCH_DIRECTIVE_pn-networkd-dhcp-conf = "Driver=smsc95xx lan78xx"
 IMAGE_INSTALL_append_sota = " virtual/network-configuration "
 
 PREFERRED_PROVIDER_virtual/bootloader_sota ?= "u-boot"
-UBOOT_ENTRYPOINT_sota ?= "0x00008000"
+UBOOT_ENTRYPOINT_sota ?= "0x00080000"
```

Comment:
Just a little bit curious about this part. I think this address should be the entry point of kernel, so it should be same as the TEXT_OFFSET in the kernel. I found the TEXT_OFFSET for rpi3-32 is 0x8000 while it's 0x80000 for rpi3-64. So, maybe use 0x8000 for 32 bit build and use 0x80000 for 64 bit build would make it even better. But the most interesting thing I think is that why the rpi3-32 still work well after this change ? Since rpi3-32 bit is already verified, maybe I made something wrong, I hope I can do some test about it later. And if you already found what's wrong in my understanding, it will be very appreciated.

Reply:
Thanks for looking into that!

Here is the source for this change: https://gitlab.denx.de/u-boot/u-boot/blob/HEAD/include/configs/rpi.h#L112. From the info I got, it actually doesn't need to match with TEXT_OFFSET exactly as we use zImage which takes care for itself of the relocation to the correct location: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/arm/booting.rst#n161

Reply:
Thank you very much !